### PR TITLE
perf(pattern-router): improve performance when create null object

### PR DESCRIPTION
**Pull request by EdamAme-x**
Reused null object

### The author should do the following, if applicable

- [ ] Add tests
- [ ] Run tests
- [ ] `bun run format:fix && bun run lint:fix` to format the code
- [ ] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

File: src/router/pattern-router/router.ts [hunk 1]
```diff
@@ -46,7 +48,7 @@ export class PatternRouter<T> implements Router<T> {
       if (routeMethod === method || routeMethod === METHOD_NAME_ALL) {
         const match = pattern.exec(path)
         if (match) {
-          handlers.push([handler, match.groups || Object.create(null)])
+          handlers.push([handler, match.groups || emptyParams])
```
Comment (yusukebe) [inline src/router/pattern-router/router.ts:51]: @EdamAme-x 

A quick question. Is there a possibility that the content of `emptyParams` will be changed? Like `emptyParams.someKey = "newValue"`.

Comment (EdamAme-x) [reply src/router/pattern-router/router.ts:51]: Since params should be abstracted by `c.req.param`, so I think there should be no such use case.

Comment (yusukebe) [reply src/router/pattern-router/router.ts:51]: Okay. Thanks!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Pull request merged**